src/app.rs

```diff
@@ -34,52 +34,37 @@ fn local_storage() -> Option<Storage> {
     }
 }
 
+const THEME_SOURCE_KEY: &str = "theme";
+const STRUCT_SOURCE_KEY: &str = "struct";
+const TMPL_SOURCE_KEY: &str = "template";
+
 fn get_data_from_local_storage(storage: &Storage, key: &str) -> Option<String> {
-    let source = storage.get_item(key).ok()??;
-    match jzon::parse(&source) {
-        Ok(jzon::JsonValue::String(source)) => Some(source),
-        _ => None,
-    }
+    storage.get_item(key).ok()?
 }
 
 #[function_component]
 pub fn App() -> Html {
     let state = use_state(|| {
-        let tmp;
-        let tmp2;
-        let (struct_source, template_source) = if let Some(storage) = local_storage() {
-            let struct_source =
-                match get_data_from_local_storage(&storage, "rinja-app-struct-source") {
-                    Some(source) => {
-                        tmp = source;
-                        &tmp
-                    }
-                    _ => STRUCT_SOURCE,
-                };
-            let template_source =
-                match get_data_from_local_storage(&storage, "rinja-app-template-source") {
-                    Some(source) => {
-                        tmp2 = source;
-                        &tmp2
-                    }
-                    _ => TMPL_SOURCE,
-                };
-            (struct_source, template_source)
-        } else {
-            (STRUCT_SOURCE, TMPL_SOURCE)
+        let local_storage = local_storage();
+        let local_storage_or = |default: &str, key: &str| -> Rc<str> {
+            let value = local_storage
+                .as_ref()
+                .and_then(|ls| get_data_from_local_storage(ls, key));
+            Rc::from(match value.as_deref().unwrap_or_default() {
+                "" => default,
```

Comment:
This one is wrong. You might have emptied the whole source code. In this case, you don't want the default to be put back.

Reply:
I thought this might be better than putting nothing in there. Will change back to restore the empty data.